Sources/TextFile.swift

```diff
@@ -274,27 +293,34 @@ open class TextFileStreamWriter: TextFileStream {
         guard let fileHandle = path.fileHandleForWriting else {
             return nil
         }
+        self.append = append
         if append {
             fileHandle.seekToEndOfFile()
         }
         super.init(fileHandle: fileHandle, delimiter: delimiter, encoding: encoding)
     }
 
-    /// Write a new line in file
-    /// - Parameter line:      the line
-    /// - Parameter delim:     append the delimiter (default: true)
-    ///
-    /// - Returns: true if successfully.
-    @discardableResult
+    /**
+     Write a new line in file
+
+     - Parameters:
+         - line: the line
+         - delim: append the delimiter (default: true)
+
+     - Throws: `FileKitError.WriteToFileFail`
+    */
     open func write(line: String, delim: Bool = true) -> Bool {
-        if let handle = fileHandle, let data = line.data(using: self.encoding) {
-            handle.write(data)
-            if delim {
-                handle.write(delimData)
-            }
-            return true
+        guard let handle = fileHandle, let data = line.data(using: self.encoding) else {
+            return false
+        }
+        if delim && append {
+            handle.write(delimData)
```

Comment:
This is a behaviour change
if append mode, you want the delim only before?
This change must be documented carefully(method comment, changelog when releasing) or be reverted